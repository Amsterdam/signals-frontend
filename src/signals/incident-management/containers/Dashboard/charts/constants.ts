// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2023 Gemeente Amsterdam

export const timeFormatLocale = {
  dateTime: '%a %e %B %Y %X',
  date: '%d-%m-%Y',
  time: '%H:%M:%S',
  periods: ['AM', 'PM'],
  days: [
    'zondag',
    'maandag',
    'dinsdag',
    'woensdag',
    'donderdag',
    'vrijdag',
    'zaterdag',
  ],
  shortDays: ['Zo', 'Ma', 'Di', 'Wo', 'Do', 'Vr', 'Za'],
  months: [
    'januari',
    'februari',
    'maart',
    'april',
    'mei',
    'juni',
    'juli',
    'augustus',
    'september',
    'oktober',
    'november',
    'december',
  ],
  shortMonths: [
    'jan',
    'feb',
    'mrt',
    'apr',
    'mei',
    'jun',
    'jul',
    'aug',
    'sep',
    'okt',
    'nov',
    'dec',
  ],
}
