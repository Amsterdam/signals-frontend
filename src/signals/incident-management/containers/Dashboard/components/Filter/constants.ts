// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2023 Gemeente Amsterdam
export const filterNames = [
  'department',
  'category',
  'priority',
  'punctuality',
  'district',
]
