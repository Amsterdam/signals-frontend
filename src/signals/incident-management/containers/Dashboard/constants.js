export const REQUEST_DASHBOARD = 'src/Dashboard/REQUEST_DASHBOARD';
export const REQUEST_DASHBOARD_SUCCESS = 'src/Dashboard/REQUEST_DASHBOARD_SUCCESS';
