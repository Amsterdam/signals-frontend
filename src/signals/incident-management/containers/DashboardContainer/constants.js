export const REQUEST_DASHBOARD = 'src/DashboardContainer/REQUEST_DASHBOARD';
export const REQUEST_DASHBOARD_SUCCESS = 'src/DashboardContainer/REQUEST_DASHBOARD_SUCCESS';
export const REQUEST_DASHBOARD_ERROR = 'src/DashboardContainer/REQUEST_DASHBOARD_ERROR';
