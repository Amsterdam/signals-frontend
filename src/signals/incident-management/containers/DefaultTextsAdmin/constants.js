export const FETCH_DEFAULT_TEXTS = 'src/DefaultTextsAdmin/FETCH_DEFAULT_TEXTS';
export const FETCH_DEFAULT_TEXTS_SUCCESS = 'src/DefaultTextsAdmin/FETCH_DEFAULT_TEXTS_SUCCESS';
export const FETCH_DEFAULT_TEXTS_ERROR = 'src/DefaultTextsAdmin/FETCH_DEFAULT_TEXTS_ERROR';

export const STORE_DEFAULT_TEXTS = 'src/DefaultTextsAdmin/STORE_DEFAULT_TEXTS';
export const STORE_DEFAULT_TEXTS_SUCCESS = 'src/DefaultTextsAdmin/STORE_DEFAULT_TEXTS_SUCCESS';
export const STORE_DEFAULT_TEXTS_ERROR = 'src/DefaultTextsAdmin/STORE_DEFAULT_TEXTS_ERROR';

export const ORDER_DEFAULT_TEXTS = 'src/DefaultTextsAdmin/ORDER_DEFAULT_TEXTS';
