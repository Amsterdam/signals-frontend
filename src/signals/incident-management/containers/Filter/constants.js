export const SAVE_FILTER = 'sia/Filter/SAVE_FILTER';
export const SAVE_FILTER_SUCCESS = 'sia/Filter/SAVE_FILTER_SUCCESS';
export const SAVE_FILTER_FAILED = 'sia/Filter/SAVE_FILTER_FAILED';

export const UPDATE_FILTER = 'sia/Filter/UPDATE_FILTER';
export const UPDATE_FILTER_SUCCESS = 'sia/Filter/UPDATE_FILTER_SUCCESS';
export const UPDATE_FILTER_FAILED = 'sia/Filter/UPDATE_FILTER_FAILED';

export const GET_FILTERS = 'sia/Filter/GET_FILTERS';
export const GET_FILTERS_SUCCESS = 'sia/Filter/GET_FILTERS_SUCCESS';
export const GET_FILTERS_FAILED = 'sia/Filter/GET_FILTERS_FAILED';

export const CLEAR_FILTER = 'sia/Filter/CLEAR_FILTER';
