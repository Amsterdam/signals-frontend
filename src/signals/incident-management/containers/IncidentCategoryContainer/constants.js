export const REQUEST_CATEGORY_UPDATE = 'sia/IncidentCategoryContainer/REQUEST_CATEGORY_UPDATE';
export const REQUEST_CATEGORY_UPDATE_SUCCESS = 'sia/IncidentCategoryContainer/REQUEST_CATEGORY_UPDATE_SUCCESS';
export const REQUEST_CATEGORY_UPDATE_ERROR = 'sia/IncidentCategoryContainer/REQUEST_CATEGORY_UPDATE_ERROR';
