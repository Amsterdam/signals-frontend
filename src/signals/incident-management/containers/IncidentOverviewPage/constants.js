export const REQUEST_INCIDENTS = 'src/OverviewPage/REQUEST_INCIDENTS';
export const REQUEST_INCIDENTS_SUCCESS = 'src/OverviewPage/REQUEST_INCIDENTS_SUCCESS';
export const REQUEST_INCIDENTS_ERROR = 'src/OverviewPage/REQUEST_INCIDENTS_ERROR';

export const INCIDENT_SELECTED = 'src/OverviewPage/INCIDENT_SELECTED';
export const FILTER_INCIDENTS_CHANGED = 'src/OverviewPage/FILTER_INCIDENTS_CHANGED';
export const PAGE_INCIDENTS_CHANGED = 'src/OverviewPage/PAGE_INCIDENTS_CHANGED';
export const SORT_INCIDENTS_CHANGED = 'src/OverviewPage/SORT_INCIDENTS_CHANGED';

export const MAIN_CATEGORY_FILTER_SELECTION_CHANGED = 'src/OverviewPage/MAIN_CATEGORY_FILTER_SELECTION_CHANGED';
