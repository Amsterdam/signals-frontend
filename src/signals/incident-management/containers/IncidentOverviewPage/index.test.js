import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import { IncidentOverviewPage, mapDispatchToProps } from './index';
import { REQUEST_INCIDENTS, INCIDENT_SELECTED, MAIN_CATEGORY_FILTER_SELECTION_CHANGED } from './constants';

import { getContext } from '../../../../../internals/testing/test-utils';

jest.mock('./components/Filter', () => () => 'Filter');
jest.mock('./components/List', () => () => 'List');
jest.mock('./components/Pager', () => () => 'Pager');

describe('<IncidentContainer />', () => {
  let props;

  beforeEach(() => {
    props = {
      overviewpage: {
        incidents: [],
        loading: false,
        filter: {},
        incidentsCount: 666,
        page: 3
      },
      onRequestIncidents: jest.fn(),
      onIncidentSelected: jest.fn(),
      onMainCategoryFilterSelectionChanged: jest.fn(),
      baseUrl: ''
    };
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  describe('rendering', () => {
    it('should render correctly', () => {
      const state = fromJS({
        global: {},
        incidentOverviewPage: {
          incidents: []
        }
      });
      const context = getContext(state);
      const wrapper = shallow(<IncidentOverviewPage {...props} />, { context });

      expect(wrapper).toMatchSnapshot();
    });

    it('should render loader correctly', () => {
      const state = fromJS({
        global: {},
        incidentOverviewPage: {
          incidents: []
        }
      });

      props.overviewpage.loading = true;

      const context = getContext(state);
      const wrapper = shallow(<IncidentOverviewPage {...props} />, { context });

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    it('should request incidents', () => {
      mapDispatchToProps(dispatch).onRequestIncidents({ filter: {}, page: 666 });
      expect(dispatch).toHaveBeenCalledWith({ type: REQUEST_INCIDENTS, payload: { filter: {}, page: 666 } });
    });

    it('should select an incident', () => {
      mapDispatchToProps(dispatch).onIncidentSelected({ id: 666 });
      expect(dispatch).toHaveBeenCalledWith({ type: INCIDENT_SELECTED, payload: { id: 666 } });
    });

    it('should change the filter', () => {
      mapDispatchToProps(dispatch).onMainCategoryFilterSelectionChanged({ category__sub: ['Graffiti / wildplak'] });
      expect(dispatch).toHaveBeenCalledWith({ type: MAIN_CATEGORY_FILTER_SELECTION_CHANGED, payload: { category__sub: ['Graffiti / wildplak'] } });
    });
  });
});
