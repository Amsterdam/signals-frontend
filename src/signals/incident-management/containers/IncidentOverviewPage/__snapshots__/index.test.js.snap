// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncidentContainer /> rendering should render correctly 1`] = `
<div
  className="overview-page"
>
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <Component
        filter={Object {}}
        onMainCategoryFilterSelectionChanged={[MockFunction]}
        onRequestIncidents={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {},
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
      />
    </div>
    <div
      className="col-9"
    >
      <div>
        <Component
          incidentSelected={[MockFunction]}
          incidents={Array []}
          incidentsCount={666}
          onRequestIncidents={
            [MockFunction] {
              "calls": Array [
                Array [
                  Object {},
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
        />
        <Component
          incidentsCount={666}
          onPageChanged={[Function]}
          page={3}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<IncidentContainer /> rendering should render loader correctly 1`] = `
<div
  className="overview-page"
>
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <Component
        filter={Object {}}
        onMainCategoryFilterSelectionChanged={[MockFunction]}
        onRequestIncidents={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {},
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
      />
    </div>
    <div
      className="col-9"
    >
      <LoadingIndicator />
    </div>
  </div>
</div>
`;
